### PR TITLE
Some work on the slider widget

### PR DESCRIPTION
**Pull request by thomascft**

### CODE REVIEW

File: lib/wibox/widget/slider.lua [hunk 1]
```diff
@@ -460,25 +462,65 @@ function slider:draw(_, cr, width, height)
...
+    -- @property bgimage
+    -- @tparam[opt=nil] image|nil bgimage
+    -- @see gears.surface
+
+    --[[ function slider:set_handle_icon(icon, ...)
+        self._private.handle_icon = type(icon) == "function" and icon or surface.load(icon)
+        self._private.handle_icon_args = { ... }
+        self:emit_signal("widget::redraw_needed")
+        self:emit_signal("property::handle_icon", icon)
+    end ]]
```
Comment (Aire-One) [inline lib/wibox/widget/slider.lua:519]: Code example doesn't need to be wrapped by block style comments. What's important is to indent them further, so ldoc recognizes them as code-block.

By convention, all ldoc block comments are still single line comments starting with `--`.

Alternatively (and preferably for this kind of feature), we write examples in the `test/examples` directory, then reference the file with a `--@DOC_path_with_underscore_EXAMPLE@` comment line. Running the `ldoc` make target (also ran by the default target) generates images from the examples and inserts them at the corresponding comments' location.

File: lib/wibox/widget/slider.lua [hunk 2]
```diff
@@ -460,25 +462,65 @@ function slider:draw(_, cr, width, height)
...
+        or nil
+
+
+    if handle_icon then
+        cr:save()
+        -- set clip
+        -- cr:clip(self._private.handle_shape(cr, handle_height, handle_width))
+        --cr:translate(230, 230)
+        cr:scale(0.25, 0.25)
+        --cr:translate(handle_width*-0.5, handle_width*-0.5)
```
Comment (Aire-One) [inline lib/wibox/widget/slider.lua:494]: Please keep your code clean.

File: lib/wibox/widget/slider.lua [hunk 3]
```diff
@@ -265,41 +267,42 @@ local slider = {mt={}}
...
     -- Handle
-    handle_shape         = shape.rectangle,
-    handle_color         = false,
-    handle_margins       = {},
-    handle_width         = false,
-    handle_border_width  = 0,
-    handle_border_color  = false,
+    handle_shape        = shape.rectangle,
+    handle_color        = false,
+    handle_icon         = false,
```
Comment (Aire-One) [inline lib/wibox/widget/slider.lua:272]: `handle_bgimage` sounds more consistent with other APIs naming

**Review by Aire-One — CHANGES_REQUESTED** [review]
Thank you for your work here!

**Comment by thomascft** [comment]
Sorry, accidental pull request.

**Pull request closed without merging**